python-cim/samples/dump_class_instance.py

```diff
@@ -60,17 +70,22 @@ def main(type_, path, namespaceName, className, key_specifier=None):
             ci = o.get_ci(namespaceName, className, instance.instance_key)
             instances.append(ci)
 
-    for instance in instances:
-        print("%s" % "=" * 80)
-        #print(compute_instance_hash(index, instance))
-        try:
-            print(dump_instance(instance, encoding='ascii', encoding_errors='ignore'))
-        except:
-            print("ERROR: failed to dump class instance!")
-            print(traceback.format_exc())
+    if csv:
+        with open(csv, 'w') as out_file:
+            out_file.write(dump_instances_csv(instances, encoding='ascii', encoding_errors='ignore'))
```

Comment:
i realize you're probably just following the existing code as an example, but i've never liked that construct, so this is an opportunity to fix things.